src/main/java/ch/njol/skript/effects/EffSpread.java

```diff
@@ -0,0 +1,114 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.Changer.ChangerUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionList;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.registrations.DefaultClasses;
+import ch.njol.skript.util.ClassInfoReference;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+@Name("Spread Across")
+@Description("A general effect to spread a list of objects across a number of settable expressions")
+@Examples({
+	"spread 1, 2, 3, 4 and 5 across {_one}, {_two}, {_three and four::*} and {_five}",
+	"spread (shuffled all players) across {_winner} and {_losers::*}",
+})
+@Since("INSERT VERSION")
+public class EffSpread extends Effect {
+
+	static {
+		Skript.registerEffect(EffSpread.class, "spread %objects% across %objects%");
+	}
+
+	private static final ClassInfoReference SINGLE_OBJECT_REFERENCE = new ClassInfoReference(DefaultClasses.OBJECT);
+
+	private Expression<?> objectsToSpread;
+	private ExpressionList<?> spreadTarget;
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		objectsToSpread = expressions[0];
+		if (objectsToSpread.isSingle()) {
+			Skript.error("You must provide more than one object to spread");
+			return false;
+		}
+		Expression<?> potentialSpreadTarget = expressions[1];
+		if (!(potentialSpreadTarget instanceof ExpressionList<?>)) {
+			Skript.error("The spread target must be a list of settable expressions");
+			return false;
+		}
+		spreadTarget = (ExpressionList<?>) potentialSpreadTarget;
+		if (!spreadTarget.getAnd()) {
+			return false;
+		}
+		if (!listChildrenCanBeSet(spreadTarget)) {
+			Skript.error("All expressions in the spread target list must be settable");
+		}
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		Object[] objectsToSpread = this.objectsToSpread.getArray(event);
+		Expression<?>[] spreadTargets = spreadTarget.getExpressions();
+		int finalObjectIndex = objectsToSpread.length - 1;
+		int objectIndex = 0;
+		for (int targetIndex = 0; targetIndex < spreadTargets.length; targetIndex++) {
+			if (objectIndex > finalObjectIndex) {
+				return;
+			}
+			Expression<?> spreadTarget = spreadTargets[targetIndex];
+			if (spreadTarget.isSingle()) {
+				spreadTarget.change(event, new Object[] { objectsToSpread[objectIndex] }, ChangeMode.SET);
+				objectIndex++;
```

Comment:
`x++` is a get-and-increment operator so you can actually do it in the place you're getting it.
```suggestion
				spreadTarget.change(event, new Object[] { objectsToSpread[objectIndex++] }, ChangeMode.SET);
```

<details>
  <summary>Fun facts about get-and-increment</summary>
  
  For any uninitiated readers, get-and-increment means we get the value and _then_ add 1 to it, as opposed to the increment-and-get operator `++x` which adds 1 to the value and _then_ gives us the result.

  If we're not needing the initial result (e.g. we're just increasing a counter) it's _technically_ more efficient to use increment-and-get because of the instruction order but the difference in today's machinery is so miniscule it wouldn't even be detectable. There's a cool argument about this on stackexchange.
  
</details>

Reply:
it was a stylistic decision - I find this clearer or less likely to be misread